K2Bridge/Utils/DateMathParser.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+namespace K2Bridge.Visitors
+{
+    using System.Text.RegularExpressions;
+
+    /// <content>
+    /// Parser for the Date Math expressions.
+    /// </content>
+    public class DateMathParser
+    {
+        public static string ParseDateMath(string expr)
+        {
+            // See Wiki or ES docs for details on date math syntax
+            var pattern = @"^([0-9-T:]+|now)\|*([0-9-+]*)([a-zA-Z]*)\/?([a-zA-Z]*)";
```

Comment:
Sadly I think that date math is more complex than that.
According to https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html#date-math
and https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-daterange-aggregation.html

It seems to support more date formats. The example they give have a "." as a seperator. I think this also has to do with the "Format" argument that the aggregation gets, but I'm not sure. Either way, this regex doesn't catch it.
My recommendation is to extract the date, and try parsing it with DateTime.Parse. And if you have a format, use that as ParseExact first.

Reply:
+1 for the "." separator.

The "format" argument is meant for formatting results and is not used by Kibana. It does not relate to the date format in the range expression. ("it is also possible to specify a date format by which the from and to response fields will be returned.")

I am relying on Kusto to parse the date string, using `make_datetime()`, so what I need is to properly extract the whole date string and pass it on. Kusto understand "." as date separator for example, so that would work (e.g. `print make_datetime('2018.02.01 13:12:13');`).

Reply:
NB: I suppose I could "steal" the Regex from the .NET client...

https://github.com/elastic/elasticsearch-net/blob/2f93905e1e8e22c3bb03bd4865ecc315fe063675/src/Nest/CommonOptions/DateMath/DateMath.cs#L25

Or even use the .NET client to split the DateMath expression components?

Reply:
If we already have a dependency on it it's probably a great idea

Reply:
We don't use it currently, we would need to add it: https://www.nuget.org/packages/NEST/

Reply:
Oh, didn't get this package was available. It could have helped before. As we are now independent, don't know if this is a good thing to introduce this dependency, it depends if using the new regex involves big changes or not. Let's you and @AsafMah decide.

Reply:
I looked at the package, but it's a pretty big dependency to take and I am not even sure their API will let us do what we want (extract the distinct parts of the date math expression). I have pushed a change to use their regex.